ReText/editor.py

```diff
@@ -370,14 +370,28 @@ def resizeEvent(self, event):
 	def highlightCurrentLine(self):
 		if globalSettings.relativeLineNumbers:
 			self.lineNumberArea.update()
-		if not globalSettings.highlightCurrentLine:
+		if globalSettings.highlightCurrentLine == 'disabled':
 			return self.setExtraSelections([])
-		selection = QTextEdit.ExtraSelection();
-		selection.format.setBackground(colorValues['currentLineHighlight'])
-		selection.format.setProperty(QTextFormat.FullWidthSelection, True)
-		selection.cursor = self.textCursor()
-		selection.cursor.clearSelection()
-		self.setExtraSelections([selection])
+		elif globalSettings.highlightCurrentLine == 'cursor-line':
+			selection = QTextEdit.ExtraSelection();
+			selection.format.setBackground(colorValues['currentLineHighlight'])
+			selection.format.setProperty(QTextFormat.FullWidthSelection, True)
+			selection.cursor = self.textCursor()
+			selection.cursor.clearSelection()
+			self.setExtraSelections([selection])
+		elif globalSettings.highlightCurrentLine == 'wrapped-line':
+			selections = [QTextEdit.ExtraSelection(),QTextEdit.ExtraSelection()]
+			selections[0].format.setBackground(colorValues['currentLineHighlight'])
+			selections[0].format.setProperty(QTextFormat.FullWidthSelection, True)
+			selections[0].cursor = self.textCursor()
+			selections[0].cursor.clearSelection()
+			selections[0].cursor.movePosition(QTextCursor.StartOfBlock)
+			selections[0].cursor.movePosition(QTextCursor.EndOfBlock, QTextCursor.KeepAnchor)
+			selections[1].format.setBackground(colorValues['currentLineHighlight'])
+			selections[1].format.setProperty(QTextFormat.FullWidthSelection, True)
+			selections[1].cursor = self.textCursor()
+			selections[1].cursor.movePosition(QTextCursor.EndOfBlock)
```

Comment:
This second selection does not actually select any text. Why is it needed?

Reply:
It is necessary to have an highlighting that extend past the end of the last line.

Reply:
It can be removed, but doing so will make the highlighting stop at the last letter of the last line. I added it because in my opinion this way it is nicer to look at, and more consistent. Do you prefer that I remove it?

Reply:
Never mind, I just found another way that doesn't need the second selection. I will remove those lines of code.

Reply:
Now it doesn't work on the last line of the document :-(

Reply:
You are right I didn't think about that... How could that be fixed?

Reply:
You can revert to your first commit. I only asked why the second selection was needed, it was not obvious :) Because it's useful, no need to remove it.